### DISCUSSION TITLE
Can not use `type` as field name of a custom type

### DISCUSSION THREAD

**Discussion by xhh** [gleam-lang/gleam#1796] [Questions & support] [answered]
```gleam
type MyType {
  Big
  Small
}

type MyData {
  MyData(type: MyType) // <--- compile error
}
```

Would it be better to allow this use case?

**Comment by lpil** [comment, marked as answer]
Hello! `type` is a keyword so it cannot be used as a variable. It is quite common for languages not to permit keywords to be used as variable names.

I tend to use `type_` in my Gleam and Rust programs intead.

**Reply by xhh** [reply]
I was trying to use it as a field name (not variable name), like field names in Elixir struct or Erlang record, which could be accessed like `my_data.type`.

**Reply by lpil** [reply]
Aye, same rules apply for labels.
